rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec

```diff
@@ -0,0 +1,115 @@
+# Software Collections. These macros are not used in default
+# Fedora builds, and should not be blindly copied or enabled.
+# Specifically, the "scl" macro must not be defined in official Fedora
+# builds. For more information, see:
+# http://docs.fedoraproject.org/en-US/Fedora_Contributor_Documentation
+# /1/html/Software_Collections_Guide/index.html
+
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name foreman_remote_execution
+
+%define rubyabi 1.9.1
+
+Summary:    Plugin that brings remote execution capabilities to Foreman
+Name:       %{?scl_prefix}rubygem-%{gem_name}
+Version:    0.0.1
+Release:    1%{?foremandist}%{?dist}
+Group:      Applications/System
+License:    GPLv3
+URL:        http://github.com/theforeman/foreman_remote_execution
+Source0:    http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires:   foreman >= 1.9.0
+Requires:   foreman < 1.10.0
+
+%if 0%{?fedora} > 18
+Requires: %{?scl_prefix}ruby(release)
+%else
+Requires: %{?scl_prefix}ruby(abi) >= %{rubyabi}
+%endif
+Requires: %{?scl_prefix}rubygems
+Requires: %{?scl_prefix}rubygem-foreman-tasks >= 0.7.3
+Requires: %{?scl_prefix}rubygem-foreman-tasks < 0.8.0
+
+%if 0%{?fedora} > 18
+BuildRequires: %{?scl_prefix}ruby(release)
+%else
+BuildRequires: %{?scl_prefix}ruby(abi) >= %{rubyabi}
+%endif
+BuildRequires: foreman-plugin >= 1.9.0
+BuildRequires: foreman-assets
+BuildRequires: %{?scl_prefix}rubygems-devel
+BuildRequires: %{?scl_prefix}rubygems
+BuildRequires: %{?scl_prefix}rubygem-foreman-tasks >= 0.7.3
+BuildRequires: %{?scl_prefix}rubygem-foreman-tasks < 0.8.0
+
+BuildArch: noarch
+
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-plugin-remote_execution
+
+%description
+A plugin bringing remote execution to the Foreman, completing the config management functionality with remote management functionality
+
+%package doc
+BuildArch:  noarch
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Documentation for rubygem-%{gem_name}
+
+%description doc
+This package contains documentation for rubygem-%{gem_name}.
+
+%prep
+%setup -n %{pkg_name}-%{version} -q -c -T
+mkdir -p .%{gem_dir}
+%{?scl:scl enable %{scl} "}
+gem install --local --install-dir .%{gem_dir} \
+            --force %{SOURCE0} --no-rdoc --no-ri
+%{?scl:"}
+
+%build
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* %{buildroot}%{gem_dir}/
+
+%foreman_bundlerd_file
+%foreman_precompile_plugin -a -s
+
+%posttrans
+# We need to run the db:migrate after the install transaction
+/usr/sbin/foreman-rake db:migrate  >/dev/null 2>&1 || :
+/usr/sbin/foreman-rake db:seed  >/dev/null 2>&1 || :
+%{foreman_apipie_cache}
+(/sbin/service foreman status && /sbin/service foreman restart) >/dev/null 2>&1
+exit 0
+
+%files
+%dir %{gem_instdir}
+%{gem_instdir}/app
+%{gem_instdir}/config
+%{gem_instdir}/db
+%{gem_instdir}/locale
+%{gem_instdir}/lib
+%{gem_instdir}/public
```

Comment:
The reason for not namespacing was that the templates STI and tooling didn't work with namespaced JobTemplate model. Since most of objects starts with Job prefix and we felt we might merge remote execution to core later we decided not to put everything under RemoteExecution namespace.